level00/PropertyTests.hs

```diff
@@ -0,0 +1,153 @@
+{-# OPTIONS_GHC -Wno-unused-binds -Wno-unused-imports #-}
+{-# LANGUAGE RankNTypes #-}
+module PropertyTests (propertyTests) where
+
+import           Control.Applicative (liftA2)
+import           Control.Lens        (Prism', matching, preview, review)
+import           Control.Lens.TH     (makePrisms)
+
+import           Data.Function       (on)
+import           Data.List           (filter, insertBy, nubBy)
+
+import           Test.Tasty          (TestTree, testGroup)
+import           Test.Tasty.Hedgehog (testProperty)
+
+import           Hedgehog
+import qualified Hedgehog.Gen        as Gen
+import qualified Hedgehog.Range      as Range
+
+import           MyBTree
+
+-- [OPTIONAL]
+import           LawPropertiesBonus  (myBTreePrismLaws)
+
+----------------------------------------------------------------------------------------------------
+addTen :: Int -> Int
+addTen = (+10)
```

Comment:
`appEndo . foldMap Endo $ replicate 10 succ`

Reply:
you're a monster... love it.